monai/transforms/compose.py

```diff
@@ -168,3 +169,101 @@ def inverse(self, data):
+            raise AssertionError("transforms and weights should be same size if both specified as sequences.")
+        self.weights = ensure_tuple(self._normalize_probabilities(weights))
+
+    def _normalize_probabilities(self, weights):
+        if len(weights) == 0:
+            return weights
+        else:
+            weights = np.array(weights)
+            if np.any(weights < 0):
+                raise AssertionError("Probabilities must be greater than or equal to zero.")
+            if np.all(weights == 0):
+                raise AssertionError("At least one probability must be greater than zero.")
+            weights = weights / weights.sum()
+            return list(weights)
+
+    def flatten(self):
+        transforms = []
+        weights = []
+        for t, w in zip(self.transforms, self.weights):
+            # if nested, probability is the current weight multiplied by the nested weights,
+            # and so on recursively
+            if isinstance(t, OneOf):
+                tr = t.flatten()
+                for t_, w_ in zip(tr.transforms, tr.weights):
+                    transforms.append(t_)
+                    weights.append(w_ * w)
+            else:
+                transforms.append(t)
+                weights.append(w)
+        return OneOf(transforms, weights, self.map_items, self.unpack_items)
+
+    def __call__(self, data):
+        if len(self.transforms) == 0:
+            return data
+        else:
+            index = self.R.multinomial(1, self.weights).argmax()
+            _transform = self.transforms[index]
+            data = apply_transform(_transform, data, self.map_items, self.unpack_items)
+            # if the data is a mapping (dictionary), append the OneOf transform to the end
+            if isinstance(data, Mapping):
+                for key in data.keys():
+                    if key + InverseKeys.KEY_SUFFIX in data:
+                        self.push_transform(data, key, extra_info={"index": index})
+            return data
+
+    def inverse(self, data):
+        if len(self.transforms) == 0:
+            return data
+        if not isinstance(data, Mapping):
+            raise RuntimeError("Inverse only implemented for Mapping (dictionary) data")
+
+        # loop until we get an index and then break (since they'll all be the same)
+        index = None
+        for key in data.keys():
+            if key + InverseKeys.KEY_SUFFIX in data:
+                # get the index of the applied OneOf transform
+                index = self.get_most_recent_transform(data, key)[InverseKeys.EXTRA_INFO]["index"]
+                # and then remove the OneOf transform
+                self.pop_transform(data, key)
+        if index is None:
+            raise RuntimeError("No invertible transforms have been applied")
+
+        # if applied transform is not InvertibleTransform, throw error
+        _transform = self.transforms[index]
+        if not isinstance(_transform, InvertibleTransform):
```

Comment:
Seems we use `warning` instead of `exception at other places:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/inverse_batch_transform.py#L47